XCUITests/PrivateBrowsingTest.swift

```diff
@@ -0,0 +1,149 @@
+        let privateHistory = app.tables["History List"].cells.count - 2
+        XCTAssertTrue(privateHistory == 1, "History entries in private browsing do not match")
+    }
+
+    func testTabCountShowsOnlyNormalOrPrivateTabCount() {
+        // Open two tabs in normal browsing and check the number of tabs open
+        navigator.openNewURL(urlString: url1)
+        navigator.goto(TabTray)
+        navigator.goto(NewTabScreen)
+        navigator.goto(TabTray)
+
+        waitforExistence(app.collectionViews.cells[url1Label])
+        let numTabs = app.collectionViews.cells.count
+        XCTAssertEqual(numTabs, 2, "The number of regular tabs is not correct")
+
+        // Open one tab in private browsing and check the total number of tabs
+        navigator.goto(NewPrivateTabScreen)
+        navigator.openURL(urlString: url2)
+        waitForValueContains(app.textFields["url"], value: "yahoo")
+
+        navigator.nowAt(NewPrivateTabScreen)
+        navigator.goto(PrivateTabTray)
+
+        waitforExistence(app.collectionViews.cells[url2Label])
+        let numPrivTabs = app.collectionViews.cells.count
+        XCTAssertEqual(numPrivTabs, 1, "The number of private tabs is not correct")
+
+        // Go back to regular mode and check the total number of tabs
+        navigator.goto(TabTray)
+
+        waitforExistence(app.collectionViews.cells[url1Label])
+        waitforNoExistence(app.collectionViews.cells[url2Label])
+        let numRegularTabs = app.collectionViews.cells.count
+        XCTAssertEqual(numRegularTabs, 2, "The number of regular tabs is not correct")
+    }
+
+    func testClosePrivateTabsOptionClosesPrivateTabs() {
+        // Check that Close Private Tabs when closing the Private Browsing Button is off by default
+        navigator.goto(SettingsScreen)
+        let appsettingstableviewcontrollerTableviewTable = app.tables["AppSettingsTableViewController.tableView"]
+        appsettingstableviewcontrollerTableviewTable.staticTexts["Firefox needs to reopen for this change to take effect."].swipeUp()
+        let closePrivateTabsSwitch = appsettingstableviewcontrollerTableviewTable.switches["Close Private Tabs, When Leaving Private Browsing"]
+
+        XCTAssertFalse(closePrivateTabsSwitch.isSelected)
+
+        // Open a Private tab
+        navigator.goto(PrivateTabTray)
+        navigator.openURL(urlString: url1)
+        navigator.nowAt(NewPrivateTabScreen)
+
+        // Go back to regular browser
+        navigator.goto(TabTray)
+
+        // Go back to private browsing and check that the tab has not been closed
+        navigator.goto(PrivateTabTray)
+        waitforExistence(app.collectionViews.cells[url1Label])
+        let numPrivTabs = app.collectionViews.cells.count
+        XCTAssertEqual(numPrivTabs, 1, "The number of tabs is not correct, the private tab should not have been closed")
+
+        // Now the enable the Close Private Tabs when closing the Private Browsing Button
+        navigator.nowAt(PrivateTabTray)
```

Comment:
No, it's been removed. In fact with the new parameter in navigator.openURL, I could remove a few navigator.nowAt, since it already said where we are